### PR TITLE
Add no doc flag and test for it

### PR DESCRIPTION
**Pull request by lydia-duncan**
This adds the ability to specify that a module, variable, function, or class or
record type should not have documentation generated for it through chpldoc, even
if the file it lives in is run through chpldoc.  This is done through the pragma
"no doc", which is attached before the symbol in question.  For example:

pragma "no doc"
class Foo {...}

will cause no documentation for the class Foo.  This is true even of functions
that Foo specified within itself which do not have this pragma guarding them.

The test case I have added ensures this works for modules, configs, classes, and
functions.  Methods and records fall out trivially.  It also ensures that what
would normally be considered a chpldoc comment will not attach to later symbols
in the case where that symbol was skipped (though I don't expect this to be an
issue).

### CODE REVIEW

**Comment by lydia-duncan** [comment]
Most of the line differences are because of some indentation changes.  There should only be a few line changes that matter, but unfortunately the diff looks a little messy.

File: test/chpldoc/nodoc/sub_test [hunk 1]
```diff
@@ -0,0 +1,6 @@
+$CHPL_HOME/util/test/sub_test $1
+touch NOEXEC
+$CHPL_HOME/util/test/sub_test $1doc
+rm NOEXEC
+
+exit $?
```
Comment (thomasvandoren) [inline test/chpldoc/nodoc/sub_test]: I think the newer format for the sub_test files (e.g. https://github.com/chapel-lang/chapel/blob/master/test/chpldoc/sub_test ) is needed here.

File: compiler/passes/docs.cpp [hunk 2]
```diff
@@ -84,7 +84,7 @@ void docs(void) {
     mkdir(folderName.c_str(), S_IWUSR|S_IRUSR|S_IXUSR);
     
     forv_Vec(ModuleSymbol, mod, gModuleSymbols) {
-      if (!devOnlyModule(mod) || developer) {
+      if ((!devOnlyModule(mod) || developer) && !mod->hasFlag(FLAG_NO_DOC)) {
```
Comment (thomasvandoren) [inline compiler/passes/docs.cpp]: I don't think it really matters, but it seems like putting the `!mod->hasFlag(FLAG_NO_DOC)` check first would take better advantage of boolean short circuiting.

**Comment by thomasvandoren** [comment]
Looks good

**Pull request merged**